src/apphosting/secrets/index.ts

```diff
@@ -3,70 +3,74 @@ import * as iam from "../../gcp/iam";
 import * as gcsm from "../../gcp/secretManager";
 import * as gcb from "../../gcp/cloudbuild";
 import * as gce from "../../gcp/computeEngine";
+import * as apphosting from "../../gcp/apphosting";
 import { FIREBASE_MANAGED } from "../../gcp/secretManager";
 import { isFunctionsManaged } from "../../gcp/secretManager";
 import * as utils from "../../utils";
 import * as prompt from "../../prompt";
 
-function fetchServiceAccounts(projectNumber: string): {
-  buildServiceAccount: string;
-  runServiceAccount: string;
-} {
-  // TODO: For now we will always return the default CBSA and CESA. When the getBackend call supports returning
-  // the attached service account in a given backend/location then return that value instead.
-  // Sample Call: await apphosting.getBackend(projectId, location, backendId); & make this function async
-  return {
-    buildServiceAccount: gcb.getDefaultServiceAccount(projectNumber),
-    runServiceAccount: gce.getDefaultServiceAccount(projectNumber),
-  };
+/** Interface for holding the service account pair for a given Backend. */
```

Comment:
While this feels cleaner, we may be overengineering here. Adding these interfaces (and the methods supporting them) will add more code surface to refactor when the backend changes land (which should be imminently). Although it isn't perfect, I would actually recommend treating run and build service accounts the same to save us from having to refactor all of these methods _again_ EOW

ie `serviceAccountsforBackend(): [string]`

Reply:
Chatted offline, we will revisit this after the demo

Reply:
Technically this makes the other suggested change redundant, but oh well. Per offline conversation we'll evaluate and change after the current stack is committed, otherwise I worry that there will be a _beast_ of a merge reconciliation in the downstream branches.

FWIW, I was worried because I wasn't sure that private build pools would bind to a specific service account. From a quick glance, it does not seem to be the case, so we can steer into just using a single list of service accounts in the future.